scipy/interpolate/interpnd.pyx

```diff
@@ -55,8 +56,8 @@ class NDInterpolatorBase:
 
     """
 
-    def __init__(self, points, values, fill_value=np.nan, ndim=None,
-                 rescale=False, need_contiguous=True, need_values=True):
+    def __init__(self, points, values=None, fill_value=np.nan, ndim=None,
+                 rescale=False, need_contiguous=True, need_values=True, xi=None):
```

Comment:
Cannot do this, the signature must be preserved. `values` need to stay a required argument, and `xi` cannot appear here. You create an interpolator with given `values` and call it with given new coordinates `xi`.

Reply:
I am quoting from my original PR:
```
The PR adds an ability to change interpolation values in an existing CloughTocher2DInterpolator object, while also saving the barycentric coordinates of interpolation points.
```
Is there a better way to do this?
This change in the API doesn't affect existing code at all, and without changing it this way, or at least with the addition of `xi=None`, there is no way to save the barycentric coordinates between runs without maintining a problematic global state to the object.